recipes/logfire/meta.yaml

```diff
@@ -0,0 +1,89 @@
+{% set name = "logfire" %}
+{% set version = "0.18.2" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/logfire-{{ version }}.tar.gz
+  sha256: ecf479246877004a94ab65668a1425a8fc42e0e2658cade180acaba9f2c78f57
+
+build:
+  entry_points:
+    - logfire = logfire.cli:main
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
+  number: 0
+
+requirements:
+  host:
+    - python >=3.8
+    - poetry-core >=1.0.0
+    - pip
+  run:
+    - python >=3.8
+    - opentelemetry-sdk >=1.21.0
+    - opentelemetry-exporter-otlp-proto-http >=1.21.0
+    - opentelemetry-instrumentation >=0.41b0
+    - rich >=13.4.2
+    - protobuf >=4.23.4
+    - typing-extensions >=4.0.0
+    - tomli >=2.0.1
+    - psycopg >=2.9.1
+  run_constrained:
```

Comment:
What's the reason for the long list here?

Reply:
I don't understand the question. Can't a package have dependency constraints?

Reply:
We have many optional dependencies.

Reply:
Yes, a package can have constraints but this seems like an unreasonably large number. This will make the installation of this package for people that don't use **all the optional** features very hard. I would propose to not handle them this way.

Reply:
@Kludex it looks like some of these requirements can be removed completely, and some moved to optional groups.

If condaforge doesn't support optional groups, just remove them.

Reply:
conda(-forge) supports optional dependencies but this way it isn't user-friendly. You're better off doing something like black does: https://github.com/conda-forge/black-feedstock/blob/main/recipe/meta.yaml

Reply:
I've simplified this. 👍